helix-core/src/main/java/org/apache/helix/task/TaskUtil.java

```diff
@@ -1040,26 +1040,39 @@ public static void purgeExpiredJobs(String workflow, WorkflowConfig workflowConf
   /**
    * The function that loops through the all existing workflow contexts and removes IdealState and
    * workflow context of the workflow whose workflow config does not exist.
+   * Try-catch has been used to avoid concurrent modification exception while doing deep copy. Since
+   * Map.keySet() can produce concurrent modification exception.
```

Comment:
1. move this to where try-catch is?
2. Explain why Map.keySet() throws CME w regards to copy constructor?

Reply:
Done. Please let me know if it is still not clear.